### ISSUE TITLE
[BUG] 200 Line Massacre always crashes on exit in latest GZDoom build (only on 4.11.0+).

### ISSUE THREAD

**Issue by andrikpowell** [ZDoom/gzdoom#2237]
### GZDoom version

GZDoom 4.11.1 / 4.11.3

### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Windows 11

### Please describe your specific OS version

Windows 11 Home (64-bit)

### Relevant hardware info

Intel i9-9900k, Geforce RTX 3080

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

When exiting the game, GZDoom hard crashes.
This does not happen while running any other WADs, even other ones by me.

Does not happen in GZDoom 4.10.0 or ZDoom / Zandronum.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. run 200lnm.wad
2. press "exit game"
3. game crashes.

### Your configuration

```ini
https://gist.github.com/andrikpowell/9e044b36f78156f96f44cae3b1909a27
```

### Provide a Log

Code: C0000005 (Access Violation - tried to read address 00000153000001a7)
Address: 00007ff740eaa117
Flags: 00000000

Full crash report: https://drive.google.com/file/d/1wAgzKefVMhcwxbB78nR_65e2W6RQoxSc/view?usp=sharing

**Comment by coelckers** [comment]
It doesn't crash for me just with this mod loaded. Please try to narrow it down to the file causing the problem. I'm fairly certain it's one of the other mods you are loading.

**Comment by andrikpowell** [comment]
> It doesn't crash for me just with this mod loaded. Please try to narrow it down to the file causing the problem. I'm fairly certain it's one of the other mods you are loading.

 I should've tested fresh to begin with tbh.
 Well I just ran from a fresh GZDoom 4.11.3 install, and I found what it doesn't like.

It is if you load both "200lnm.wad" and "200lnm.deh" it was cause a crash when exiting. Previous 4.10.0 doesn't crash when loading both. So it must have something to do with the Dehacked parser.

**Comment by coelckers** [comment]
fixed in f1017ef362831db87b5b4b1cf7f202063e119ec5
